ruby/ql/lib/codeql/ruby/frameworks/http_clients/NetHttp.qll

```diff
@@ -30,20 +37,27 @@ class NetHttpRequest extends Http::Client::Request::Range instanceof DataFlow::C
     |
       // Net::HTTP.get(...)
       method in ["get", "get_response"] and
-      requestNode = API::getTopLevelMember("Net").getMember("HTTP").getReturn(method) and
+      connectionNode = API::getTopLevelMember("Net").getMember("HTTP") and
+      requestNode = connectionNode.getReturn(method) and
```

Comment:
You can move `requestNode = connectionNode.getReturn(method)` up after line 36, and then remove the three duplicated lines.

Reply:
It's curious to me that you can set `requestNode` before you've set `connectionNode` in this situation. I've made the requested change, but I'd like to learn more. Is there some documentation you can point me to describing this behavior, or is that "just the way it works" with queries?

Reply:
QL is a logic language, so you should not think of `x = y` as an assignment, but rather as a constraint (`x` must be equal to `y`). So the order doesn't really matter.